canonical/maas-ui
# update compose VM side panel

## Done

- update compose VM side panel layout to fit in a side panel
- add ShowAdvanced component
- simplify to use a single button for all breakpoints
- add workaround for vanilla tooltip bug https://github.com/canonical/vanilla-framework/issues/4607
- update stepper layout to match designs

This will be followed up by https://warthogs.atlassian.net/browse/MAASENG-1265

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

Make sure you can compose VMs and have Virsh/LXD setup on your machine

- Go to KVM -> LXD and select a host
- Click "Add VM"
- Define interfaces
- Make sure all input fields work correctly including contextual menus
- Add disk
- Make sure multiple disks can be added and deleted (1 disk is required)
- Make sure you can submit the form successfully and the data sent over the websocket matches the form

## Fixes

Fixes: MAASENG-1252

## Launchpad issue

Related Launchpad maas issue in the form `lp#number`.

## Backports

In general, please propose fixes against _main_ rather than release branches (e.g. 2.7), unless the fix is only applicable for that specific release. Please apply backport labels to the PR (e.g. "Backport 2.7") for the appropriate releases to target.

Only bug and security fixes should be backported, new features should only land in main.

## Screenshots
### After
#### Show advanced transition
![Kapture 2023-01-24 at 09 03 05](https://user-images.githubusercontent.com/7452681/214240236-cfe61bac-eb93-474d-afea-ed9c1e95d05a.gif)

#### Subnet select dropdown
![image](https://user-images.githubusercontent.com/7452681/214257772-5017cf45-6e39-4a7f-87b1-3d05e5e26275.png)

#### Interfaces
![image](https://user-images.githubusercontent.com/7452681/214257947-f01bde43-f29f-4958-b0eb-6eb98b9052e2.png)

#### Single buttons
![image](https://user-images.githubusercontent.com/7452681/214257699-1d0882b5-2c32-4b9c-b52f-8c85d89f08c3.png)

### New stepper layout

#### Before
<img width="533" alt="Pasted Graphic 2" src="https://user-images.githubusercontent.com/7452681/214258190-0eff5daf-ca71-4b6d-85df-5e8f65463874.png">

#### After
<img width="533" alt="Pasted Graphic 1" src="https://user-images.githubusercontent.com/7452681/214258309-095c3800-e44c-4bcb-a2c4-19ed0c38aa33.png">
